src/main/java/com/stirante/runechanger/sourcestore/impl/LolalyticsSource.java

```diff
@@ -0,0 +1,200 @@
+                    biggestValRune = entry;
+                }
+            }
+            r.getRunes().add(Objects.requireNonNull(biggestValRune).getKey());
+        }
+        // SECONDARY RUNES
+
+        Map.Entry<Rune, int[]> biggestValSecondaryRune = null;
+        for (Map.Entry<Rune, int[]> entry : convertedDataPair.getRuneDataConverted().get("rune2").entrySet()) {
+            if (biggestValSecondaryRune == null || entry.getValue()[mode] > biggestValSecondaryRune.getValue()[mode]) {
+                biggestValSecondaryRune = entry;
+            }
+        }
+        final Style secondaryStyle = Objects.requireNonNull(biggestValSecondaryRune).getKey().getStyle();
+        final int secondaryUsedSlot = biggestValSecondaryRune.getKey().getSlot();
+        r.setSubStyle(secondaryStyle);
+        r.getRunes().add(biggestValSecondaryRune.getKey());
+
+        Map<Rune, int[]> remainingSecondaryRunes = convertedDataPair.getRuneDataConverted().get("rune2").entrySet().stream()
+                .filter(map -> map.getKey().getStyle() == secondaryStyle && map.getKey().getSlot() != secondaryUsedSlot)
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        Map.Entry<Rune, int[]> biggestRemainingSecondaryRune = null;
+        for (Map.Entry<Rune, int[]> entry : remainingSecondaryRunes.entrySet()) {
+            if (biggestRemainingSecondaryRune == null || entry.getValue()[mode] > biggestRemainingSecondaryRune.getValue()[mode]) {
+                biggestRemainingSecondaryRune = entry;
+            }
+        }
+
+
+        r.getRunes().add(Objects.requireNonNull(biggestRemainingSecondaryRune).getKey());
+
+        //MODIFIERS
+
+
+        for (int i = 3; i < 6; i++) {
+            Map<Modifier, int[]> modifierList = convertedDataPair.getModifierDataConverted().get("rune" + i);
+            Map.Entry<Modifier, int[]> biggestValModifier = null;
+            for (Map.Entry<Modifier, int[]> entry : modifierList.entrySet()) {
+                if (biggestValModifier == null || entry.getValue()[mode] > biggestRemainingSecondaryRune.getValue()[mode]) {
+                    biggestValModifier = entry;
+                }
+            }
+            r.getModifiers().add(Objects.requireNonNull(biggestValModifier).getKey());
+        }
+
+        if (!r.verify()) {
+            log.error("Failed to verify the runepage.");
+            return null;
+        }
+        return r;
+
+    }
+
+    @Override
+    public void getRunesForChampion(Champion champion, GameMode mode, ObservableList<RunePage> pages) {
+        if(champion != null) {
+            downloadRunes(champion, pages);
+        }
+    }
+
+    @Override
+    public String getSourceName() {
+        return "Lolalytics";
+    }
+
+    @Override
+    public String getSourceKey() {
+        return "lolalytics.com";
+    }
+
+    private static class LolalyticsResult {
+        private Display display;
```

Comment:
You can change `Display display` to `Map<String, Map<String, int[]>> display`. This way you can iterate over it, since a little bit above, you repeat 2 lines 3 times.